src/tests/Model/ResultSummaryCreatorTests.cs

```diff
@@ -0,0 +1,162 @@
+﻿using NUnit.Framework;
+using System;
+
+namespace NUnit.Gui.Model
+{
+    [TestFixture]
+    public class ResultSummaryCreatorTests
+    {
+        [Test]
+        public void WhenResultNodeIsNotForTestRunExceptionIsThrown()
+        {
+            Assert.That(() => CreateResultSummary("<anything-other-than-test-run/>"), Throws.InstanceOf<InvalidOperationException>());
+
+            Assert.That(() => CreateResultSummary("<test-run/>"), Throws.Nothing);
+        }
+        
+        [Test]
+        public void WhenResultIsNotSpecified_PassedIsDefault()
+        {
+            var summary = CreateResultSummary("<test-run/>");
+
+            Assert.That(summary.OverallResult, Is.EqualTo("Passed"));
+
+            summary = CreateResultSummary("<test-run result='Failed'/>");
+
+            Assert.That(summary.OverallResult, Is.EqualTo("Failed"));
+        }
+
+        [Test]
+        public void WhenDurationIsNotSpecified_ZeroIsDefault()
+        {
+            var summary = CreateResultSummary("<test-run/>");
+
+            Assert.That(summary.Duration, Is.EqualTo(0.0));
+
+            summary = CreateResultSummary("<test-run duration='1.9'/>");
+
+            Assert.That(summary.Duration, Is.EqualTo(1.9));
+        }
+
+        [Test]
+        public void WhenStartTimeIsNotSpecified_DateTimeMinValueIsDefault()
+        {
+            var summary = CreateResultSummary("<test-run/>");
+
+            Assert.That(summary.StartTime, Is.EqualTo(DateTime.MinValue));
+
+            var expectedDate = new DateTime(2017, 7, 8, 6, 19, 23);
+            summary = CreateResultSummary($"<test-run start-time='{expectedDate.ToString("u")}'/>");
+
+            Assert.That(summary.StartTime, Is.EqualTo(expectedDate));
+        }
+
+        [Test]
+        public void WhenEndTimeIsNotSpecified_DateTimeMaxValueIsDefault()
+        {
+            var summary = CreateResultSummary("<test-run/>");
+
+            Assert.That(summary.EndTime, Is.EqualTo(DateTime.MaxValue));
+
+            var expectedDate = new DateTime(2017, 7, 8, 6, 21, 46);
+            summary = CreateResultSummary($"<test-run end-time='{expectedDate.ToString("u")}'/>");
+
+            Assert.That(summary.EndTime, Is.EqualTo(expectedDate));
+        }
+
+        [Test]
+        public void TestCountIsCountOfEachNestedTestCase()
+        {
+            var innerXml =
+                "<test-case/>" +
+                "<test-case/>" +
+                "<test-suite>" +
+                    "<test-case/>" +
+                    "<test-case/>" +
+                "</test-suite>";
+            var summary = CreateResultSummary($"<test-run>{innerXml}</test-run>");
+
+            Assert.That(summary.TestCount, Is.EqualTo(4));
+        }
+
+        [Test]
+        public void WhenNoResultIsNotSpecifiedInTestCase_SkipCountIsIncremented()
```

Comment:
Remove Not from test name. E.g.
WhenNoResultIsNotSpecifiedInTestCase_SkipCountIsIncremented => WhenNoResultIsSpecifiedInTestCase_SkipCountIsIncremented